cpp/HelloWorld/client/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.22)
+project(armonik_cpp_hello_client VERSION 0.1 LANGUAGES C CXX)
+
+find_package(Protobuf REQUIRED)
+find_package(gRPC CONFIG REQUIRED)
+find_package(Threads)
+find_package(ArmoniK.Api.Common REQUIRED)
+find_package(ArmoniK.Api.Client REQUIRED)
+
+set(SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src")
+
+
+file(GLOB_RECURSE SRC_FILES ${SRC_DIR}/*.cpp)
+file(GLOB_RECURSE HEADER_FILES ${SRC_DIR}/*.h)
+
+add_executable(${PROJECT_NAME} ${SRC_FILES} ${HEADER_FILES})
+
+set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 14)
+
+target_link_libraries(${PROJECT_NAME} PUBLIC ArmoniK.Api.Common ArmoniK.Api.Client)
+
+target_include_directories(${PROJECT_NAME} PUBLIC ${SRC_DIR} "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>")
+
+
+install(TARGETS ${PROJECT_NAME}
+        EXPORT ${PROJECT_NAME}Targets
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        )
```

Comment:
Also I think you need a CMakeLists fille in the project root and that cmake will call the client and worker cmake depending on the options

Reply:
I already have a Makefile for building the images, each Dockerfile runs the cmake build with the relevant CMakeLists.txt. Having an additional CMakeLists in the root directory will add unneeded complexity (the Dockerfile will call a CMakeLists.txt which will call with a second CMakeLists.txt) with no apparent added benefit. It'll just obfuscate what is otherwise a very clear/direct build process. 

In my mind, the samples exist to illustrate how ArmoniK can be used. So they should be as clear as possible in terms of how one can build an ArmoniK application.

Reply:
Yes I saw the Makefile but the comment was more about the C++ project construction. Also I'm not a fan of using a makefile to run cmake build. But you can keep it that way for now